src/main/java/step2/StringCalculator.java

```diff
@@ -0,0 +1,82 @@
+package step2;
+
+public class StringCalculator {
+
+    interface Calculator {
+        int calculate(int result, int nextNumber);
+    }
+
+    enum Operator {
+
+        PLUS("+", (result, nextNumber) -> result + nextNumber),
+        MINUS("-", (result, nextNumber) -> result - nextNumber),
+        MULTIPLY("*", (result, nextNumber) -> result * nextNumber),
+        DIVIDE("/", (result, nextNumber) -> result / nextNumber);
+
+        private String operator;
+        private Calculator calculator;
+
+        Operator(String operator, Calculator calculator) {
+            this.operator = operator;
+            this.calculator = calculator;
+        }
+
+        public static int calculate(int result, int nextNumber, String operator) {
+            for (Operator op : Operator.values()) {
+                if (op.operator.equals(operator)) {
+                    return op.calculator.calculate(result, nextNumber);
+                }
+            }
+            throw new IllegalArgumentException();
+        }
```

Comment:
이 메소드는 String operator에 대응하는 Operator enum 값을 찾고,
해당 enum의 calculator를 호출하는,
두 가지 역할을 가지고 있네요

두 개의 역할을 분리하는 건 어떨까요?
```suggestion
        public Operator of(String operator) {
            for (Operator op : Operator.values()) {
                if (op.operator.equals(operator)) {
                    return op;
                }
            }
            throw new IllegalArgumentException();
        }
        
        public int calculate(int result, int nextNumber) {
            return calculator.calculate(result, nextNumber);
        }
```

Reply:
// calculate
        for (int i = 0; i < inputArray.length - 1; i += 2) {
            result = Operator.of(inputArray[i + 1]).calculate(result, Integer.parseInt(inputArray[i + 2]));
        }
                  와 !! 이렇게 분리가 또 가능하네요 감사합니다 ^^